Updating documentation around deploy markers (#9359)
circleci/circleci-docs: jekyll/_cci2/deploy/deploys-overview.adoc

[#introduction]
== Introduction

CircleCI deploys gives you visibility into your Kubernetes deployments within the CircleCI web app.

> When setting up deploys you can choose to rely on a Release Agent that gets installed in your cluster or you can instead choose an agentless approach.

Reviewer: ```suggestion
When setting up deploys you can choose one of two options:

* *Release agent*: Choose to use the CircleCI release agent, which you install in your cluster to enable you to manage your deployments through CircleCI.
* *Agentless*: Add CLI commands to your config to enable logging and tracking of deploys and deployment status in the CircleCI UI.
```

> If you choose the *agent-based* approach, you will have access to the following deploy management tools from CircleCI:

Reviewer: ```suggestion
If you choose the release agent approach, you will have access to the following deploy management tools from CircleCI:
```

* Restore version
* Scale component
* Restart component

> The agent based approach supports link:https://argoproj.github.io/argo-rollouts/[Argo Rollouts] for progressive delivery. Currently the link:https://argo-rollouts.readthedocs.io/en/stable/concepts/#canary[canary deployment strategy] is supported, but in a future release link:https://argo-rollouts.readthedocs.io/en/stable/concepts/#blue-green[blue-green] will also be supported. Some further controls are available for projects that use Argo Rollouts:

Reviewer: ```suggestion
The release agent supports link:https://argoproj.github.io/argo-rollouts/[Argo Rollouts] for progressive delivery. Currently the link:https://argo-rollouts.readthedocs.io/en/stable/concepts/#canary[canary deployment strategy] is supported, but in a future release link:https://argo-rollouts.readthedocs.io/en/stable/concepts/#blue-green[blue-green] will also be supported. Some further controls are available for projects that use Argo Rollouts:
```
